### PR TITLE
Add `allowemailregister` option

### PR DESCRIPTION
**Pull request by SISheogorath**
A small contribution to solve some "paper cut issues" ;)

#297

### CODE REVIEW

**Comment by jackycute** [comment]
@SISheogorath You are so nice!
Could you also add description to `README.md`?

**Comment by SISheogorath** [comment]
Done

**Comment by jackycute** [comment]
I know, we should write some tests #22

**Comment by jackycute** [comment]
Thank you @SISheogorath !

**Comment by SISheogorath** [comment]
Ehhh >.> did you test that it works? My last test ended up in a setting `HMD_ALLOW_EMAIL_REGISTER` and not showing the button.

**Comment by jackycute** [comment]
You mean there might have so bugs?
I could revert it if needed, and you can patch it up.
I might need some time to test it.

File: lib/config.js [hunk 1]
```diff
@@ -132,6 +132,7 @@ if (process.env.HMD_LDAP_PROVIDERNAME) {
 }
 var imgur = process.env.HMD_IMGUR_CLIENTID || config.imgur || false;
 var email = process.env.HMD_EMAIL ? (process.env.HMD_EMAIL === 'true') : !!config.email;
+var allowemailregister = process.env.HMD_ALLOW_EMAIL_REGISTER ? (process.env.HMD_HMD_ALLOW_EMAIL_REGISTER === 'true') : !!config.allowemailregister;
```
Comment (jackycute) [inline lib/config.js]: Well, here is the bug.
This should be:
```
var allowemailregister = process.env. HMD_ALLOW_EMAIL_REGISTER ? (process.env. HMD_ALLOW_EMAIL_REGISTER === 'true') : ((typeof config. allowemailregister === 'boolean') ? config. allowemailregister : true);
```

Comment (jackycute) [inline lib/config.js]: And there is a typo right there `HMD_HMD_ALLOW_EMAIL_REGISTER`.

Comment (jackycute) [inline lib/config.js]: Above fixed in 6be875263a3c0ee849b023983032d8f55e28a2e0

**Comment by jackycute** [comment]
I just made a patch @SISheogorath, please help to test.

**Comment by SISheogorath** [comment]
Cool. I'll test it in a few seconds ;)

**Comment by SISheogorath** [comment]
@jackycute Works great! Thanks!

**Comment by youngcraft** [comment]
how to use it in dockerfile version? I use docker-compose to build my Codimd. In Readme.md , there is not reference issues to talk about it.

**Pull request merged**